src/components/user/auth/signupModal.js

```diff
@@ -0,0 +1,301 @@
+import React, { Component } from "react";
+import { Container, Row, Col, Modal, Form, Button } from "react-bootstrap";
+import { connect } from "react-redux";
+import { signup } from "../../../reducers/authSlice";
+import { checkFieldValidation } from "../../../commonFunctions/validateFormField";
+
+class SignUpModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: "",
```

Comment:
If possible try to minimise the stuff in state

Reply:
This was a struggle for me as well. We can have a single error field but then the submit button disable feature won't work (which is saving a bunch of server computation by avoiding any incomplete requests). Anything you can think of to be reduced?